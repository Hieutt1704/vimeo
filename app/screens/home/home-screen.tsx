import { StackScreenProps } from "@react-navigation/stack"
import { observer } from "mobx-react-lite"
import React, { FC } from "react"
import { FlatList, View } from "react-native"
import { AutoImage, Header, Screen, Text } from "../../components"
import { NavigatorParamList } from "../../navigators"
import homeScreenStyles from "./home-screen.styles"
// import { useNavigation } from "@react-navigation/native"
import { useGetCategories } from "./useGetCategories"

const BACKGROUND_IMAGE =
  "https://i.vimeocdn.com/video/1486234694-2789256b3e16a01f78abaebcc3eeb1ef872b63bc8bfda834c8c4cc6b11fb578b-d"

export const HomeScreen: FC<StackScreenProps<NavigatorParamList, "home">> = observer(
  function HomeScreen() {
    const { categories } = useGetCategories()

    // const navigation = useNavigation()

    return (
      <Screen testID="screen" style={homeScreenStyles.container} preset="fixed">
        <Header
          testID="header"
          headerTx="home.title"
          style={homeScreenStyles.header}
          titleStyle={homeScreenStyles.titleHeader}
        />
        <FlatList
          testID="list"
          data={categories}
          numColumns={2}
          keyExtractor={(item) => item.key}
          contentContainerStyle={homeScreenStyles.flatList}
          renderItem={({ item }) => (
            <View testID="item" style={homeScreenStyles.categoryContainer}>
              <AutoImage
                source={{ uri: BACKGROUND_IMAGE }}
                style={homeScreenStyles.categoryBackground}
              />
              <View style={homeScreenStyles.categoryIcon}>
                <AutoImage source={{ uri: item.image }} style={homeScreenStyles.categoryIcon} />
                <Text style={homeScreenStyles.categoryName}>{item.name}</Text>
              </View>
            </View>
          )}
          ItemSeparatorComponent={() => <View style={homeScreenStyles.categorySeparator} />}
        />
      </Screen>
    )
  },
)
