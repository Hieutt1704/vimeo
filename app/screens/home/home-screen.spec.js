import React from "react"
import { render, screen } from "@testing-library/react-native"
import { HomeScreen } from "./home-screen"

/**
 * Test View
 */
test("render correctly", () => {
  const { getByTestId } = render(<HomeScreen />)
  expect(getByTestId("screen")).toBeTruthy()
  expect(getByTestId("header")).toBeTruthy()
  expect(getByTestId("list")).toBeTruthy()
})

test("match snapshot", () => {
  render(<HomeScreen />)
  expect(screen.toJSON()).toMatchSnapshot()
})
