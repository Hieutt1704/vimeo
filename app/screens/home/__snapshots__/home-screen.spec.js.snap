// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "height": "100%",
        },
        Object {},
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="screen"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
        },
        Object {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 24,
            "paddingHorizontal": 16,
            "paddingTop": 24,
          },
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CDD4DA",
            "paddingTop": 12,
          },
        ]
      }
      testID="header"
    >
      <View
        style={
          Object {
            "width": 32,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Helvetica",
                "fontSize": 15,
              },
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "#1d1d1d",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ],
            ]
          }
        >
          home.title.test
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 32,
          }
        }
      />
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      contentContainerStyle={
        Object {
          "paddingVertical": 8,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="list"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
